src/Plugin.php

```diff
@@ -626,7 +621,13 @@ public function checkPluginState($plugin_key)
                 );
             }
 
-            $this->update($input);
+            // Update status and all plugin info in case it was changed in a new update (like the license)
+            $this->update(
+                [
+                    'id'    => $plugin->fields['id'],
+                    'state' => self::NOTUPDATED,
```

Comment:
I do not understand why state should be changed to `NOTUPDATED` when previous state was `ANEW` or `NOTINSTALLED`.